### PR TITLE
Bugfix: Complete the bugfix in from adj matrices for time series graph

### PR DESCRIPTION
**Pull request by matteo-causalens**
## Description and Motivation
The bugfix in https://github.com/causalens/cai-causal-graph/pull/60 still had an issue.

## How Has This Been Tested?

New unit tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor (moving code around, general refactoring improvements)
- [ ] Documentation (documentation)

## PR Checklist:

- [x] I have reviewed my own PR? (review the PR as you are reviewing someone else's PR)
- [x] I have implemented all requirements? (see JIRA, project documentation)
- [x] I am affecting someone else's work? If so: I have added those people as reviewers?
- [ ] I have added comments to all the bits that are hard to follow
- [ ] At a bare minimum, I tested this manually
- [x] I have added unit test(s)
- [x] Is this a bugfix? If so have I added a regression test?
- [x] Does this PR satisfy the [one PR, one concern](https://medium.com/@fagnerbrack/one-pull-request-one-concern-e84a27dfe9f1) rule?
- [ ] If needed, documentation has been added/updated?
- [x] I have updated the appropriate changelog with a line for my changes
- [ ] If this PR breaks reproducibility, have I added a note in the changelog explaining the break in reproducibility
      and its extent?

### CODE REVIEW

File: cai_causal_graph/time_series_causal_graph.py [hunk 1]
```diff
@@ -1106,13 +1106,14 @@ def from_adjacency_matrices(
...
         time_delta_to_index = {time_delta: i for i, time_delta in enumerate(adjacency_matrices)}
+        n_time_delta = len(adjacency_matrices)
 
         for time_delta, adjacency_matrix in adjacency_matrices.items():
             for row, column in zip(*numpy.where(adjacency_matrix)):
                 # add 1 to the row and column to account for the time delta
                 adjacency_matrix_full[
-                    time_delta_to_index[time_delta] + ((shape[0] - 1) * row),
-                    time_delta_to_index[0] + ((shape[0] - 1) * column),
+                    time_delta_to_index[time_delta] + (n_time_delta * row),
```
Comment (steven-causalens) [inline cai_causal_graph/time_series_causal_graph.py:1115]: So the issue was that the shape was incorrect, i.e. one too little?

File: changelog.md [hunk 2]
```diff
@@ -6,6 +6,8 @@
   `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph` where the method was not handling properly
   undirected edges in the adjacency matrices.
 - Added `mypy-extensions` dependency as `"^1.0.0"` and moved `mypy` back to dev dependency (setting it as `"^1.8.0"`).
+- Fixed a bug in `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.from_adjacency_matrices` introduced in
+  the previous bug fix where the method was not handling properly undirected edges in the adjacency matrices.
```
Comment (steven-causalens) [inline changelog.md:10]: Say "in the `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.from_adjacency_matrices` method of `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph`, introduced ..."

**Review by steven-causalens — APPROVED** [review]
Thanks @matteo-causalens! Have you checked that this fixes tests in `cai-causal-discovery`?

File: cai_causal_graph/time_series_causal_graph.py (hunk 1 above)
Comment (matteo-causalens) [reply cai_causal_graph/time_series_causal_graph.py:1115]: the issue was that I don't have to consider the shape as "jump" but the number of lags

File: tests/test_time_series_graph.py [hunk 3]
```diff
@@ -581,6 +581,44 @@ def test_from_adjacency_matrices(self):
                 self.assertLess(source.time_lag, destination.time_lag)
                 self.assertEqual(edge.get_edge_type(), EdgeType.DIRECTED_EDGE)
 
+        # test for the bugfix
```
Comment (andrew-causalens) [inline tests/test_time_series_graph.py:584]: Can you explain a bit more what the bug was. Once this is merged no one will know what bugfix is being referred to. Why do these edge pairs act as a regression case.

File: tests/test_time_series_graph.py [hunk 4]
```diff
@@ -581,6 +581,44 @@ def test_from_adjacency_matrices(self):
...
+
+        # get the adjacency matrices from the tsdag
+        adj_matrices = tsdag.adjacency_matrices
+
+        # create the graph from the adjacency matrices
+        tsdag_2 = TimeSeriesCausalGraph.from_adjacency_matrices(adj_matrices, tsdag.variables)
+
+        # check that the graphs are equal (we need to do`get_minimal_graph` to remove the floating nodes)
+        self.assertEqual(tsdag, tsdag_2.get_minimal_graph())
+
```
Comment (andrew-causalens) [inline tests/test_time_series_graph.py:621]: Let's also do self.assertDictEqual(tsdag_2.adjacency_matrices, adj_matrices)`. Would suggest doing this check throughout this unit test for various times we move to/from matrices to graph objects.

File: cai_causal_graph/time_series_causal_graph.py (hunk 1 above)
Comment (AJM-causaLens) [reply cai_causal_graph/time_series_causal_graph.py:1116]: This fixes regime shift tests in `causal_model_inspector`, thanks Matteo!

**Review by AJM-causaLens — APPROVED** [review]

File: tests/test_time_series_graph.py [hunk 5]
```diff
@@ -581,6 +581,44 @@ def test_from_adjacency_matrices(self):
                 self.assertLess(source.time_lag, destination.time_lag)
                 self.assertEqual(edge.get_edge_type(), EdgeType.DIRECTED_EDGE)
 
+        # test for the bugfix
+        edge_pairs = [
```
Comment (maxelliott-causalens) [inline tests/test_time_series_graph.py:585]: This may as well be a separate unit test as it is entirely separated from the previous, and keeps it more of a 'unit' test

**Review by steven-causalens — APPROVED** [review]
LGTM, thanks!

**Review by steven-causalens — APPROVED** [review]

**Review by andrew-causalens — APPROVED** [review]
